In [1]:
import pandas as pd

# Creating Some Common Functions

# 1. Function to check for null (missing) values in the DataFrame

In [2]:
def check_null(df):
    data = df
    null_values =  df.isna().sum()
    return null_values

# 2. Function to check for duplicate values in the DataFrame

In [3]:
def check_dup(df):
    data = df
    dup =  len(df[df.duplicated()])
    return dup

# 3. Function to drop duplicate values in the DataFrame

In [4]:
def drop_dup(df):
    df.drop_duplicates(inplace=True)
    
    return df

# Cleaning Data 


1. Cleaning user_nodes 
    
    Checking null Values

In [24]:
df =  pd.read_csv('row_data/user_nodes.csv')
null = check_null(df)
null


id_           0
area_id_      0
node_id_      0
act_date      0
deact_date    0
has_loan      0
is_act        0
dtype: int64

Here is no null values so we didn't need to do operations on values

Checking Duplicate Values

In [6]:
dup =  check_dup(df)
dup

143

here is 143 duplicate values so remove these values 

In [7]:
df =  drop_dup(df)


In [8]:
dup =  check_dup(df)
dup

0

Successfully removed duplicated values

performing other operations 

In [9]:
df.columns

Index(['id_', 'area_id_', 'node_id_', 'act_date', 'deact_date', 'has_loan',
       'is_act'],
      dtype='object')

renaming columns names for make relationship with other tables

In [25]:
df.rename(columns = {'id_':'consumer_id','area_id_':'region_id'},  inplace=True)

In [26]:
df.columns

Index(['consumer_id', 'region_id', 'node_id_', 'act_date', 'deact_date',
       'has_loan', 'is_act'],
      dtype='object')

Deleting columns which is not required 

In [12]:
del df['has_loan']


Save the cleaned data for SQL 

In [13]:
df.to_csv('cleaned DataBase/user_nodes_cleaned.csv',index=False)

# Working on another file


In [14]:
df =  pd.read_csv('row_data/user_transactions.csv')
null = check_null(df)
null


id_                0
t_date             0
t_type             0
t_amt              0
has_credit_card    0
account_type       0
dtype: int64

Here is no null values so we didn't need to do operations on values

Checking Duplicate Values

In [15]:
dup =  check_dup(df)
dup

79

here is 79 duplicate values so remove these values

In [16]:
df =  drop_dup(df)


In [17]:
dup =  check_dup(df)
dup

0

Successfully removed duplicated values

performing other operations


In [18]:
df.columns

Index(['id_', 't_date', 't_type', 't_amt', 'has_credit_card', 'account_type'], dtype='object')


renaming columns names for make relationship with other tables



In [19]:
df.rename(columns = {'id_':'consumer_id','t_date':'transaction_date','t_type':'transaction_type','t_amt':'transaction_amount'},  inplace=True)


In [20]:
df.to_csv('cleaned DataBase/user_transaction_cleaned.csv', index=False)

# Working on another file

In [21]:
df =  pd.read_csv('row_data/world_regions.csv')
null = check_null(df)
null

region_id      0
region_name    0
dtype: int64

Here is no null values so we didn't need to do operations on values

Checking Duplicate Values

In [22]:
dup =  check_dup(df)
dup

0

here is no duplicate values 

In [23]:
df.columns

Index(['region_id', 'region_name'], dtype='object')

Here is columns names are same as we need so ignore another operations